library/src/main/kotlin/com/connectrpc/MethodSpec.kt

```diff
@@ -29,12 +24,12 @@ internal object Method {
  * @param requestClass The Kotlin Class for the request message.
  * @param responseClass The Kotlin Class for the response message.
  * @param idempotency The declared idempotency of a method.
- * @param method The HTTP method of a request.
+ * @param streamType The method's stream type.
  */
 class MethodSpec<Input : Any, Output : Any>(
     val path: String,
     val requestClass: KClass<Input>,
     val responseClass: KClass<Output>,
     val idempotency: Idempotency = Idempotency.UNKNOWN,
-    val method: String = Method.POST_METHOD,
+    val streamType: StreamType = StreamType.UNKNOWN,
```

Comment:
This unknown value seems a little scary. How will it impact things? Is it necessary for backwards-compatibility with older generated code? What could go wrong if older code is paired with this newer runtime -- would it mean that the GET HTTP verb is never used and interceptors would see the wrong value for stream type? I think it might be safer to make this a non-optional parameter with no default and remove the `UNKNOWN` sentinel enum entry. Users must re-generate their code when they update to newer version of the library.

Reply:
I went back and forth on this. Having UNKNOWN also matches connect-go and grpc-java. connect-swift doesn't have an unknown type.

It was less about the compatibility with existing generated code as this is still pre-v1 and it should be expected that consumers keep the runtime libraries in sync with the generated code.

Reply:
I know connect-go has this because there's no real support for enums in Go, and it's just good practice for the default/zero value to be a sentinel of this sort. But in kotlin and java, it's much less clear why you'd include one. With grpc-java, it's likely because that supports server-side, too, and if you want to create an "unknown method handler", it's most correct for it to report to handler/interceptor code that the stream type is unknown.